Dopamine/Dopamine.swift

```diff
@@ -0,0 +1,279 @@
+        :param: token Provided on your Dopamine dashboard
+        :param: versionID A unique string representing the application version (such as CFBundleVersion or the build number)
+        :param: userIdentity An array of String to String pairings that uniquely identify the user
+        :param: pairings An ActionPairings object representing all possible action pairings for this build
+    */
+    public init(appID: String, apiKey : String, token : String, versionID : String, userIdentity: [[String:String]], pairings : ActionPairings) {
+        s = Settings(appID, apiKey, token, versionID, pairings)
+        identity = userIdentity
+        basePayload = [
+            "key": s.apiKey,
+            "token": s.token,
+            "versionID" : s.versionID,
+            "build" : s.pairings.buildID,
+            "ClientOS" : os,
+            "ClientVersion": osVersion,
+            "ClientAPIVersion": bundleVersion ?? "Invalid Version Number",
+            "identity": userIdentity
+        ]
+    }
+    
+    /**
+        Initializes the system with Dopamine and informs Dopamine about the desired (action -> [reward,feedback]) pairings
+    */
+    public func initialize() {
+        let payload : APIPayload = [
+            "feedbackFunctions": s.pairings.feedbackFunctions,
+            "rewardFunctions": s.pairings.rewardFunctions,
+        ]
+        request(.Init, payload: payload)
+    }
+    
+    /**
+    */
+    public func reinforce(action : String, callback : ReinforcementCallback) {
+        let payload : APIPayload = [
+            "eventName": action
+        ]
+        request(.Reinforce, payload: payload, callback: callback)
+    }
+    
+    /**
+    */
+    public func track(action: String) {
+        let payload : APIPayload = [
+            "eventName": action
+        ]
+        request(.Track, payload: payload)
+    }
+    
+    /**
+        Handles actually making a request to Dopamine
+    */
+    func request(endpoint : APIEndpoint, payload : APIPayload, callback : ReinforcementCallback? = nil) {
+        let utc = NSDate().timeIntervalSince1970 * 1000.0
+        let offset = Double(NSTimeZone.localTimeZone().secondsFromGMT * 1000)
+        let local = utc + offset
+        let t : APIPayload = [
+            "UTC": utc.description,
+            "local": local.description
+        ]
+        let standardizedPayload = basePayload.merge(t)
+        let requestPayload = standardizedPayload.merge(payload)
+        let url = buildEndpointURL(endpoint)
+        Alamofire.request(.POST, url, parameters: requestPayload, encoding: .JSON)
+                 .responseJSON { _, netResponse, JSON, _ in
+                    if endpoint == .Reinforce {
+                        if let r = netResponse, j: AnyObject = JSON, functionName = j["reinforcementFunction"] as? String, c = callback {
+                            c(r.statusCode, functionName)
+                        }
+                    }
```

Comment:
@DavidAntaramian You *may* want to have a callback that happens when an error occurs (just so that the UI isn't waiting for response on something for forever)